omnigan/trainer.py

```diff
@@ -1569,17 +1613,21 @@ def masker_s_loss(self, x, z, target, domain, for_="G"):
                         full_loss += gp_loss
                     else:
                         raise NotImplementedError
-
+        if self.opts.gen.m.use_spade:
+            return full_loss, softmax(pred, dim=1).detach()
         return full_loss
 
-    def masker_m_loss(self, x, z, target, domain, for_="G"):
+    def masker_m_loss(self, x, z, target, domain, for_="G", cond=None):
         assert for_ in {"G", "D"}
         assert domain in {"r", "s"}
         self.assert_z_matches_x(x, z)
         assert x.shape[0] == target.shape[0] if target is not None else True
         full_loss = torch.tensor(0.0, device=self.device)
         # ? output features classifier
-        pred_logits = self.G.mask(z=z, sigmoid=False)
+        if self.opts.gen.m.use_spade:
+            pred_logits = self.G.decoders["m"](z, cond)
+        else:
+            pred_logits = self.G.decoders["m"](z)
```

Comment:
I updated the API to the non-spade masker so that we don't have to care about this